src/riak_kv_vnode.erl

```diff
@@ -1512,7 +1512,7 @@ prepare_put_new_object(#state{idx =Idx} = State,
                      false ->
                          []
                  end,
-    {EpochId, State2} = new_key_epoch(State),
+    {EpochId, State2} = maybe_new_key_epoch(Coord, State, undefined, RObj),
```

Comment:
Instead of using `undefined` here, could we use a more distinctive atom like `no_object` or something along those lines? Would make the code a bit clearer, and also help guard against any future bugs where an unrelated use of `undefined` might leak into this code. Admittedly this is somewhat of a personal style preference though, so I won't argue if you'd strongly prefer leave it as-is. Looking through the code, we'd have to update `maybe_new_key_epoch` and also `highest_actor`, but I think it should be a safe and easy change.

(Commenting on the first commit where this appears, but it does look like this code later got moved into `maybe_update_vclock`.)